File: zuzubibi/2월/2월 1일/달리기 경주.py
def solution(players, callings):
    answer = []
    turns = {}
    cnt = 1
    for i in range(len(callings)):
        if callings[i-1] == callings[i]:
            print(callings[i], cnt)
            cnt += 1
        else:
            print(callings[i], cnt)
            turns[callings[i]] = cnt
            cnt = 1
    
    print(turns)        
        # inx = players.index(calling)
        # players[inx-1],players[inx] = players[inx], players[inx-1]
    
    # answer = players
    # 음.. 더 고민해야겠다..
    return answer


# ["mumu", "kai", "mine", "soe", "poe"]
print(solution(["mumu", "soe", "poe", "kai", "mine"],["kai", "kai", "mine", "mine"]))

File: wooseongshin/2월/1일/달리기경주.java
import java.util.HashMap;
import java.util.*;

class Solution {
    public String[] solution(String[] players, String[] callings) {
        String[] answer = {};
        HashMap<String, Integer> map = new HashMap<>();

        for (int i = 0; i < players.length; i++) {
            map.put(players[i], i);
        }

        for(String calling: callings) {
            Integer i = map.get(calling);

            String frontPlayer = players[i-1];
            players[i-1] = calling;
            map.put(calling, i-1);

            players[i] = frontPlayer;
            map.put(frontPlayer, i);

        }


        return players;
    }
}
